### PR TITLE
~Tweak the main README~ Tweak the docs

### PR DESCRIPTION
**Pull request by coopergillan**
### What

* Use terraform flavor of markdown and split examples out
* Tweak some spelling, punctuation, and headings
* Add and correct some links, both external and within the project

### Why

* The examples will be easier to read with syntax highlighting
* To make it a little easier to read
* To add some more information

@raymondberg

### CODE REVIEW

**Comment by raymondberg** [comment]
While you work on those terraform snippets can you convert them to the new notation without quotes around the identifiers?

**Comment by coopergillan** [comment]
@raymondberg - check out 6675055 for the initial feedback.

73c5505 is to fix what I think is a syntax error. Could you confirm?

File: README.md [hunk 1]
```diff
@@ -29,106 +29,118 @@ Download for amd64 (for other architectures and OSes you can build from source a
...
 ## Examples:
 
-Provider configuration
+### Provider configuration
 
-```
+```terraform
 provider redshift {
-  "url" = "localhost",
+  url = "localhost",
```
Comment (raymondberg) [inline README.md]: The commas are all passé as well. Terraform is very hip now.

**Review by raymondberg — APPROVED** [review]
I'd strike the commas, but otherwise amazing.

**Pull request merged**